src/controllers/TalksController.php

```diff
@@ -123,37 +133,307 @@ public function postAction($request, $db) {
+
+        $isAdmin   = $talk_mapper->thisUserHasAdminOn($talk_id);
+        //$isSpeaker = $talk_mapper->thisUserIsSpeakerOn($talk_id);
+        //if (! $isAdmin  && ! $isSpeaker) {
+        if (! $isAdmin) {
+            throw new Exception('You are not entitled to edit this entry', 403);
+        }
+
+        $eUri = explode('/', $existing_talk['event_uri']);
+        $existing_talk['event_id'] = $eUri[(count($eUri)-1)];
+
+        $talk = array(
+            'talk_id' => $talk_id,
+            'event_id' => $existing_talk['event_id']
+        );
+
+        $talk['talk_title'] = filter_var(
+            $request->getParameter("talk_title"),
+            FILTER_SANITIZE_STRING
+        );
+        if(empty($talk['talk_title'])) {
+            $errors[] = "'title' is a required field";
+        }
+
+        $talk['talk_description']  = filter_var(
+            $request->getParameter("talk_description"),
+            FILTER_SANITIZE_STRING
+        );
+        if (empty($talk['talk_description'])) {
+            $errors[] = "'description' is a required field";
+        }
+
+        $talk['language'] = filter_var($request->getParameter('language'), FILTER_SANITIZE_STRING);
+        if (empty($talk['language'])) {
+            $talk['language'] = 'English - UK';
+        }
+
+        $language_mapper = new LanguageMapper($db, $request);
+        if (! $language_mapper->isLanguageAvailable($talk['language'])) {
+            $errors[] = sprintf('The language "%s" isn\'t known', $talk['language']);
+        }
+
+        $talk['slides_link'] = filter_var($request->getParameter('slides_link'), FILTER_SANITIZE_URL);
+
+        if (! $isAdmin) {
+            $talk['type']       = $existing_talk['type'];
+            $talk['duration']   = $existing_talk['duration'];
+            $talk['start_date'] = (new \DateTime($existing_talk['start_date']))->format('U');
+        } else {
+
+            $talk['type'] = filter_var(
+                $request->getParameter("type"),
+                FILTER_SANITIZE_STRING
+            );
+            if (empty($talk['type'])) {
+                $errors[] = "'type' is a required field";
+            }
+
+            if (! in_array($talk['type'], $talk_mapper->getCategories())) {
+                $errors[] = sprintf(
+                    'The talk-category "%s" isn\'t known',
+                    $talk['type']
+                );
+            }
+
+            $talk['start_date'] = (new \DateTime($request->getParameter("start_date")))->format('U');
+
+            $talk['duration'] = filter_var(
+                $request->getParameter('duration'),
+                FILTER_SANITIZE_NUMBER_INT
+            );
+
+            if (empty($talk['duration'])) {
+                $talk['duration'] = 60;
+            }
         }
+
+        if ($errors) {
+            throw new Exception(implode(". ", $errors), 400);
+        }
+
+        $talk_mapper->edit($talk, $talk_id);
+
+        header("HTTP/1.1 204 No Content", NULL, 204);
```

Comment:
You can't be guaranteed that HTTP/1.1 was the protocol used to make the request; can you make use of `http_response_code` instead?